tests/nested/manual/fde-on-classic/task.yaml

```diff
@@ -80,15 +84,29 @@ execute: |
       # Check that no reboot has been scheduled, then force a reboot
       remote.exec not test -f /run/systemd/shutdown/scheduled
 
-      remote.exec sudo reboot || true
-      tests.nested wait-for reboot "$boot_id"
-      remote.exec sudo snap watch "$REMOTE_CHG_ID"
+      if [ "$finish_install" = "true" ]; then
+          remote.exec sudo reboot || true
+          tests.nested wait-for reboot "$boot_id"
+          remote.exec sudo snap watch "$REMOTE_CHG_ID"
+      fi
   }
 
-  refresh_snap_and_reboot pc-kernel-new.snap pc-kernel
+  current_kernel_file=$(remote.exec readlink /run/mnt/ubuntu-boot/EFI/ubuntu/kernel.efi)
+  current_kernel_file=${current_kernel_file%/*}
+
+  # Test aborting a kernel installation before a reboot
+  refresh_rebooting_snap pc-kernel-new.snap pc-kernel false
```

Comment:
(nitpick^2) I think I would slightly prefer a more expressive parameter here, maybe something like this?:
```sh
index b8e46caedf..dd9eadb122 100644
--- a/tests/nested/manual/fde-on-classic/task.yaml
+++ b/tests/nested/manual/fde-on-classic/task.yaml
@@ -84,7 +84,7 @@ execute: |
       # Check that no reboot has been scheduled, then force a reboot
       remote.exec not test -f /run/systemd/shutdown/scheduled
 
-      if [ "$finish_install" = "true" ]; then
+      if [ "$finish_install" != "no-reboot ]; then
           remote.exec sudo reboot || true
           tests.nested wait-for reboot "$boot_id"
           remote.exec sudo snap watch "$REMOTE_CHG_ID"
@@ -95,7 +95,7 @@ execute: |
   current_kernel_file=${current_kernel_file%/*}
 
   # Test aborting a kernel installation before a reboot
-  refresh_rebooting_snap pc-kernel-new.snap pc-kernel false
+  refresh_rebooting_snap pc-kernel-new.snap pc-kernel no-reboot
   remote.exec sudo snap abort "$REMOTE_CHG_ID"
   # Make sure everything is in the same state
   not remote.exec stat /run/mnt/ubuntu-boot/EFI/ubuntu/try-kernel.efi
@@ -103,10 +103,10 @@ execute: |
   remote.exec cat /run/mnt/ubuntu-boot/EFI/ubuntu/grubenv | MATCH -E "^kernel_status=$"
 
   # Test successful kernel update
-  refresh_rebooting_snap pc-kernel-new.snap pc-kernel true
+  refresh_rebooting_snap pc-kernel-new.snap pc-kernel with-reboot
 
   echo "Refresh pc gadget and assert assets got updated"
-  refresh_rebooting_snap pc-new.snap pc true
+  refresh_rebooting_snap pc-new.snap pc with-reboot
   for f in /boot/grub/kernel.efi /run/mnt/ubuntu-boot/EFI/boot/bootx64.efi /run/mnt/ubuntu-boot/EFI/boot/grubx64.efi; do
       remote.exec sudo grep -q -a '"This program cannot be run in XXX mode"' "$f"
   done
```

Reply:
Sure, makes sense. I've pushed now something in this line and rewritten a bit history so it is mergeable.